src/dos/drive_fat.cpp

```diff
@@ -775,8 +775,8 @@ fatDrive::fatDrive(const char *sysFilename,
 	created_successfully = (diskfile != nullptr);
 	if (!created_successfully)
 		return;
-	fseek(diskfile, 0L, SEEK_END);
-	filesize = (uint32_t)ftell(diskfile) / 1024L;
+	cross_fseeko(diskfile, 0L, SEEK_END);
+	filesize = (uint32_t)cross_ftello(diskfile) / 1024L;
```

Comment:
(use file-size call).

I guess `filesize` is in KB - maybe lift `1024L` to a `constexpr auto bytes_per_kbyte`?